nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeRecord.java

```diff
@@ -151,7 +153,7 @@
         .description("If specified, two FlowFiles will be binned together only if they have the same value for "
             + "this Attribute. If not specified, FlowFiles are bundled by the order in which they are pulled from the queue.")
         .required(false)
-        .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
```

Comment:
does it make any sense to change this on a flowfile level? I see no reason/use case for this

Reply:
It is done in the same way in this processor's brother `MergeContent`, I think it could be useful to create different bins based on different attributes at the same time, without having to add an `UpdateAttribute` node before the `MergeRecord`. 
Whatever the case, I also think it would be more intuitive for the user to have similar processors to be aligned

Reply:
>  I also think it would be more intuitive for the user to have similar processors to be aligned

This is a good point indeed. I'll leave to others the decision on this.